### DISCUSSION TITLE
Custom login page

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#3930] [General]
Is there a way to create a custom login page? For example, one that includes images, instructions, etc. I've been trying different approaches, but every time I add the Login module to a page, many problems arise. All menus stop working, it's not possible to access the admin panel or edit existing modules on the page. The only way to regain control is by opening another page, accessing the control panel from there, and deleting the Login module using the page manager.

In any case, once this custom login page is successfully created, how can Oqtane be configured to use it by default instead of the original one?

Here's an idea of what I would like to achieve:
![image](https://github.com/oqtane/oqtane.framework/assets/103559699/70c0cfec-eaf6-4fcf-ba30-363b9a485981)

I have also tried creating my own Login module, but I can't seem to make Oqtane use it when a user is not authenticated.

**Comment by sbwalker** [comment]
@LuisVilaplana There are many ways to accomplish this. The Login page is simply a route of /login and you are free to include whatever modules you want on the page. 

To do this using the UI you can login and go to Admin Dashboard / Pages Management. Find the Login page in the list and click Edit. Select the Modules tab and delete the User Login module from the page. Now navigate to the Login page and use the Control Panel to add your custom Login module to it. 

You will want to make sure your custom Login module implements similar functionality as the default Login module. It may be useful if Oqtane separated some of url parameter functionality to a dedicated module without a UI in the future to make it simpler. 

You can also create a custom Site Template by following the examples in the core framework (Oqtane.Server\Infrastructure\SiteTemplates\DefaultSiteTemplate.cs) which use your custom Login module rather than the default Login module.

**Reply by LuisVilaplana** [reply]
Thanks Shaun, I'll give it a try

**Reply by sbwalker** [reply]
@LuisVilaplana I made a small change in 5.1 so that the Administrator is able navigate to the /Login page when they are signed in (without being redirected to the home page) so that they can manage the page content using the Control Panel)
